docs/syntax.bnf

```diff
@@ -14,6 +14,7 @@
 <path> ::= <qid>
 
 <term_id> ::= <id>
+            | (<id>)
```

Comment:
This file is automatically generated by `make doc` from the Menhir grammar. So it should not be modified by hand. So thanks to revert your changes.

Reply:
We ought to add a warning at the top of the file.

Reply:
With something like this
```diff
diff --git a/docs/Makefile b/docs/Makefile
index 792f3cdb..b24e87b0 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -21,7 +21,9 @@ help:

 .PHONY: syntax.bnf
 syntax.bnf:
-       echo "<qid> ::= [<uid>.]+<uid>\n" > $@
+       echo "; DO NOT EDIT THIS FILE MANUALLY" > $@
+       echo "; It is generated automatically with 'make syntax.bnf'." >> $@
+       echo "<qid> ::= [<uid>.]+<uid>\n" >> $@
        echo "<switch> ::= on\n           | off\n" >> $@
        echo "<assoc> ::= left\n          | right\n" >> $@
        $(BNF_GEN) ../src/parsing/lpParser.mly | sed \
@@ -116,7 +118,10 @@ syntax.bnf:

 .PHONY: syntax_dedukti.bnf
 syntax_dedukti.bnf:
-       echo "<qid> ::= [<uid>.]+<uid>\n" > $@
+       echo "; DO NOT EDIT THIS FILE MANUALLY" > $@
+       echo "; It is generated automatically with \
+'make syntax_dedukti.bnf'." >> $@
+       echo "<qid> ::= [<uid>.]+<uid>\n" >> $@
        echo "<assert> ::= #ASSERT\n           | #ASSERTNOT\n" >> $@
        $(BNF_GEN) ../src/parsing/dkParser.mly | sed \
        -e 's/ASSERT/<assert>/g' \
```

Reply:
On my machine, echo was printing the \n literally; on the internet, people are mostly against using echo and backslash-escaped characters, so I replaced them with the more universal printf